### ISSUE TITLE
require$$2.bind is not a function after vite build, works during vite dev

### ISSUE THREAD

**Issue by boid-com** [ipfs/js-ipfs#4343]
- **Version**:
`
{version: '0.18.0', commit: '', repo: '12', ipfs-core: '0.18.0', interface-ipfs-core: '^0.158.0'}
`

- **Platform**:
Chrome, Macos

#### Severity:
  Critical - System crash, application panic.

#### Description:

- What you did
 include ipfs-core in a quasar vite project
- What happened
 works during development but not after build
- What you expected to happen
should operate the same during dev and build

#### Steps to reproduce the error:

- include ipfs-core in a quasar vite project
- yarn quasar build
- try and visit the built site
- site won't render due to ipfs error\

## Error trace
```
TypeError: require$$2.bind is not a function
    at ipfs.8f5ba8b6.js:17:314645
y @ index.cff37d5b.js:556
(anonymous) @ index.cff37d5b.js:556
Promise.catch (async)
$ @ index.cff37d5b.js:556
N @ index.cff37d5b.js:556
install @ index.cff37d5b.js:556
use @ index.cff37d5b.js:1
EM @ index.cff37d5b.js:556
await in EM (async)
(anonymous) @ index.cff37d5b.js:556
Promise.then (async)
(anonymous) @ index.cff37d5b.js:556
Promise.then (async)
(anonymous) @ index.cff37d5b.js:556
```

**Comment by alan890104** [comment]
I solved this problem by upgrading vite to 4.3.3.
Here is [my repo](https://github.com/Solratic/sharex-interface).

**Comment by SgtPooki** [comment]
Hello @boid-com,

We regret to inform you that the issue you reported will not be resolved in js-ipfs as we are in the process of deprecating js-ipfs. However, it is highly likely that the issue has been addressed in our new library, [helia](https://github.com/ipfs/helia). Although we haven't explicitly tested this particular issue, [helia](https://github.com/ipfs/helia) is designed to improve upon the limitations of js-ipfs. Our migration guide is available at https://github.com/ipfs/helia/wiki/Migrating-from-js-IPFS.

We kindly request you to attempt using https://github.com/ipfs/helia to address your specific use case. If you encounter any further issues or require additional assistance, please feel free to open an issue in the helia repository.

We appreciate your understanding and cooperation during this transition period. Thank you for considering helia as a potential solution.

Best regards,
@ipfs/helia-dev
